staging/src/k8s.io/apimachinery/pkg/runtime/scheme.go

```diff
@@ -163,22 +163,13 @@ func (s *Scheme) AddUnversionedTypes(version schema.GroupVersion, types ...Objec
 // the struct becomes the "kind" field when encoding. Version may not be empty - use the
 // APIVersionInternal constant if you have a type that does not have a formal version.
 func (s *Scheme) AddKnownTypes(gv schema.GroupVersion, types ...Object) {
-	if len(gv.Version) == 0 {
```

Comment:
It is not gone, only comes after the reflect.Ptr check. And it's skipped if `types` is empty. You want to get it back here so badly?